crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -507,6 +507,9 @@ impl IdentityManager {
                                 if result.any_cleared() {
                                     changed_identity = Some((*private_identity).clone());
                                     info!(cleared = ?result, "Removed some or all of our private cross signing keys");
+                                } else if new && private_identity.has_master_key().await {
```

Comment:
I started writing unit tests for this, but the existing `test_manager_own_key_query_response` is not actually testing our user, but rather the other user (this is a bug I think). When swapped it does not work properly because we hardcode device keys into the response but create a new account from scratch, which will generate incompatible device keys. Not yet sure how to insert the relevant keys into the response instead of hardcoding them.

Reply:
There's `PrivateCrossSigningIdentity::as_upload_request()` which will let you get the public parts of the identity. Those can then be massaged into a `/keys/query` response, once that gets fed into the state machine the relevant code paths should be hit.

Reply:
hm `UploadSigningKeysRequest` does not contain the device identity though, which is the bit that is mismatched

Reply:
Maybe I could plug in something like `let keys: Raw<DeviceKeys> = account.device_keys().await.to_raw();` into the `KeyQueryResponse`?

Reply:
Hmm, why are device keys important for this? But yeah you could put the device keys via the account into the response as well.

Reply:
It is important that they match, otherwise the device from the keys/query will not be inserted into the store as per this [code block](https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk-crypto/src/identities/manager.rs#L264-L277)

Reply:
Yes, it is important that they match if you try to insert them. What I'm asking is why would you need to insert them in the first place since the unit test is solely about user identities and cross signing.

Reply:
Ah perhaps you want to check that the whole trust chain now works, instead of just checking that the identity has been marked as verified?

Reply:
Hm yea its not directly relevant to my change after all. Added it anyway for convenience into the new `key_query` helper.